ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/OcmIntegrationForm.tsx

```diff
@@ -0,0 +1,210 @@
+                }),
+        }),
+        ocm: yup.object().shape({
+            endpoint: yup.string().trim().required('Endpoint is required'),
+        }),
+        skipTestIntegration: yup.bool(),
+    }),
+    updatePassword: yup.bool(),
+});
+
+export type CloudSourceIntegrationFormValues = {
+    cloudSource: CloudSourceIntegration;
+    updateCredentials: boolean;
+};
+export const defaultValues: CloudSourceIntegrationFormValues = {
+    cloudSource: {
+        id: '',
+        name: '',
+        type: 'TYPE_OCM',
+        credentials: {
+            secret: '',
+        },
+        skipTestIntegration: true,
+        ocm: {
+            endpoint: '',
+        },
+    },
+    updateCredentials: true,
+};
+
+function OcmIntegrationForm({
+    initialValues = null,
+    isEditable = false,
+}: IntegrationFormProps<CloudSourceIntegration>): ReactElement {
+    const formInitialValues = { ...defaultValues, ...initialValues };
+    if (initialValues) {
+        formInitialValues.cloudSource = merge({}, formInitialValues.cloudSource, initialValues);
+        formInitialValues.cloudSource.credentials.secret = '';
+        formInitialValues.updateCredentials = false;
+    }
+    const {
+        values,
+        touched,
+        errors,
+        dirty,
+        isValid,
+        setFieldValue,
+        handleBlur,
+        isSubmitting,
+        isTesting,
+        onSave,
+        onTest,
+        onCancel,
+        message,
+    } = useIntegrationForm<CloudSourceIntegrationFormValues>({
+        initialValues: formInitialValues,
+        validationSchema,
+    });
+
+    const { isCreating } = usePageState();
+
+    function onChange(value, event) {
+        return setFieldValue(event.target.id, value);
+    }
+
+    function onUpdateCredentialsChange(value, event) {
+        setFieldValue('cloudSource.credentials.secret', '');
+        return setFieldValue(event.target.id, value);
+    }
+
+    return (
+        <>
+            <PageSection variant="light" isFilled hasOverflowScroll>
+                <FormMessage message={message} />
+                <Form isWidthLimited>
+                    <FormLabelGroup
+                        isRequired
+                        label="Integration name"
+                        fieldId="cloudSource.name"
+                        touched={touched}
+                        errors={errors}
+                    >
+                        <TextInput
+                            isRequired
+                            type="text"
+                            id="cloudSource.name"
+                            value={values.cloudSource.name}
+                            onChange={onChange}
+                            onBlur={handleBlur}
+                            isDisabled={!isEditable}
+                        />
+                    </FormLabelGroup>
+                    <FormLabelGroup
+                        isRequired
+                        label="OCM endpoint"
```

Comment:
I think the "OCM" is redundant here. But looking at other integrations, we have both variants already (e.g. "Registry endpoint" for Artifact and just "Endpoint" for ECR).

Reply:
Unless it is ambiguous to primary audience for this integration, just **Endpoint** seems like a solid pattern.